src/Microsoft.IdentityModel.Tokens/Validation/Validators.Audience.cs

```diff
@@ -75,92 +69,45 @@ internal static AudienceValidationResult ValidateAudience(IEnumerable<string> au
                     new ExceptionDetail(
                         new MessageDetail(
                             LogMessages.IDX10206,
                             null),
                         typeof(SecurityTokenInvalidAudienceException),
                         new StackFrame(true)));
 
-            string? validAudience = AudienceIsValidReturning(audiencesAsList, validationParameters);
+            string? validAudience = AudienceIsValidReturning((List<string>)audiences, validationParameters);
             if (validAudience != null)
             {
                 return new AudienceValidationResult(validAudience);
             }
 
             return new AudienceValidationResult(
-                    Utility.SerializeAsSingleCommaDelimitedString(audiencesAsList),
+                    Utility.SerializeAsSingleCommaDelimitedString(audiences),
                     ValidationFailureType.AudienceValidationFailed,
                     new ExceptionDetail(
                         new MessageDetail(
                             LogMessages.IDX10214,
-                            LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(audiencesAsList)),
-                            LogHelper.MarkAsNonPII(validationParameters.ValidAudience ?? "null"),
+                            LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(audiences)),
                             LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(validationParameters.ValidAudiences))),
                         typeof(SecurityTokenInvalidAudienceException),
                         new StackFrame(true)));
         }
 
-        private static bool AudienceIsValid(List<string> audiences, TokenValidationParameters validationParameters)
-        {
-            return AudienceIsValidReturning(audiences, validationParameters) != null;
-        }
-
-        private static string? AudienceIsValidReturning(List<string> audiences, TokenValidationParameters validationParameters)
+        private static string? AudienceIsValidReturning(List<string> audiences, ValidationParameters validationParameters)
         {
             string? validAudience = null;
-            if (!string.IsNullOrWhiteSpace(validationParameters.ValidAudience))
-                validAudience = AudiencesMatchSingle(audiences, validationParameters.ValidAudience, validationParameters.IgnoreTrailingSlashWhenValidatingAudience);
-
-            if (validAudience == null && validationParameters.ValidAudiences != null)
-            {
-                if (validationParameters.ValidAudiences is not List<string> validAudiences)
-                    validAudiences = validationParameters.ValidAudiences.ToList();
 
-                validAudience = AudiencesMatchList(audiences, validAudiences, validationParameters.IgnoreTrailingSlashWhenValidatingAudience);
-            }
+            List<string> validAudiences = validationParameters.ValidAudiences.ToList();
```

Comment:
ValidationParameters.ValidAudiences is an IList<string>, no need to call ToList().

Reply:
Good catch, updated. Thanks!